src/main/java/org/voltdb/exportclient/KinesisFirehoseExportClient.java

```diff
@@ -0,0 +1,264 @@
+import org.voltdb.common.Constants;
+import org.voltdb.export.AdvertisedDataSource;
+
+import au.com.bytecode.opencsv_voltpatches.CSVWriter;
+
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.RegionUtils;
+import com.amazonaws.services.kinesisfirehose.AmazonKinesisFirehoseClient;
+import com.amazonaws.services.kinesisfirehose.model.DescribeDeliveryStreamRequest;
+import com.amazonaws.services.kinesisfirehose.model.DescribeDeliveryStreamResult;
+import com.amazonaws.services.kinesisfirehose.model.InvalidArgumentException;
+import com.amazonaws.services.kinesisfirehose.model.PutRecordBatchRequest;
+import com.amazonaws.services.kinesisfirehose.model.PutRecordBatchResponseEntry;
+import com.amazonaws.services.kinesisfirehose.model.PutRecordBatchResult;
+import com.amazonaws.services.kinesisfirehose.model.Record;
+import com.amazonaws.services.kinesisfirehose.model.ResourceNotFoundException;
+import com.amazonaws.services.kinesisfirehose.model.ServiceUnavailableException;
+import com.google_voltpatches.common.base.Throwables;
+import com.google_voltpatches.common.util.concurrent.ListeningExecutorService;
+
+public class KinesisFirehoseExportClient extends ExportClientBase {
+    private static final VoltLogger m_logger = new VoltLogger("ExportClient");
+
+    String m_streamName = null;
+    AmazonKinesisFirehoseClient m_firehoseClient = null;
+    boolean m_skipInternals;
+    ExportDecoderBase.BinaryEncoding m_binaryEncoding;
+    char m_seperator;
+    // use thread-local to avoid SimpleDateFormat thread-safety issues
+    ThreadLocal<SimpleDateFormat> m_ODBCDateformat;
+
+    @Override
+    public void configure(Properties config) throws Exception
+    {
+        String regionName = config.getProperty("region","").trim();
+        if (regionName.isEmpty()) {
+            throw new IllegalArgumentException("KinesisFirehoseExportClient: must provide an region");
+        }
+        Region region = RegionUtils.getRegion(regionName);
+
+        m_streamName = config.getProperty("stream.name","").trim();
+        if (m_streamName.isEmpty()) {
+            throw new IllegalArgumentException("KinesisFirehoseExportClient: must provide a stream.name");
+        }
+
+        String accessKey = config.getProperty("access.key","").trim();
+        if (accessKey.isEmpty()) {
+            throw new IllegalArgumentException("KinesisFirehoseExportClient: must provide an access.key");
+        }
+        String secretKey = config.getProperty("secret.key","").trim();
+        if (secretKey.isEmpty()) {
+            throw new IllegalArgumentException("KinesisFirehoseExportClient: must provide a secret.key");
+        }
+        m_firehoseClient = new AmazonKinesisFirehoseClient(
+                new BasicAWSCredentials(accessKey, secretKey));
+        m_firehoseClient.setRegion(region);
+
+        m_skipInternals = Boolean.parseBoolean(config.getProperty("skipinternals", "false"));
```

Comment:
don't even give them the option to skip internals. simply do it. all new connectors have harcode skip internals to true